__tests__/app.test.js

```diff
@@ -157,3 +157,152 @@ describe('GET /api/articles', () =>
+    body, votes, author, article_id, created_at`, () =>
+    {
+        const userCommentReference = 
+            {
+                username: 'icellusedkars',
+                body: 'Fruit pastilles'
+            }
+        return request(app).post('/api/articles/1/comments').expect(201).send(userCommentReference)
+        .then(({ body }) =>
+        {
+            expect(body).toHaveProperty('body')
+            expect(body).toHaveProperty('votes')
+            expect(body).toHaveProperty('author')
+            expect(body).toHaveProperty('article_id')
+            expect(body).toHaveProperty('created_at')
+        })
+    })
+    test(`takes an object containing a username and a body in the request body and returns an object that 
+    has the same body, same username as author, and the correct article_id`, () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'Fruit pastilles'
+        }
+        return request(app).post('/api/articles/1/comments').expect(201).send(userCommentReference)
+        .then(({ body }) =>
+        {
+            expect(body.body).toEqual(userCommentReference.body)
+            expect(body.author).toEqual(userCommentReference.username)
+            expect(body.article_id).toEqual(1)
+        })
+    })
+    test('responds with error 400 if anything except a number is requested as a parameter', () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'Fruit pastilles'
+        }
+        return request(app).post('/api/articles/test/comments').expect(400).send(userCommentReference)
+        .then((err) =>
+        {
+            expect(err.body.msg).toEqual('invalid article ID: not a number')
+        })
+    })
+    test('responds with error 404 if URL specifies an ID that does not exist', () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'icellusedkars',
+            body: 'Fruit pastilles'
+        }
+        return request(app).post('/api/articles/102738619863/comments').expect(404).send(userCommentReference)
+        .then((err) =>
+        {
+            expect(err.body.msg).toEqual('invalid article ID: ID not found')
+        })
+    })
+    test('if the username that is not matched anywhere, will return an error 404', () =>
+    {
+        const userCommentReference = 
+        {
+            username: 'THIS_USER_DOES_NOT_EXIST',
+            body: 'Fruit pastilles'
+        }
+        return request(app).post('/api/articles/1/comments').expect(404).send(userCommentReference)
+        .then((err) =>
+        {
+            expect(err.body.msg).toEqual('invalid username: user not found')
+        })
+    })
+    test('if the username is matched but the reference body is not matched anywhere, will return an error 404', () =>
```

Comment:
The body won't exist. This is supposed to be a post request where you INSERT INTO the comments database

Reply:
i thought task 7 we are trying to get the relevant comment using an object containing a body for a username and body, not actually implement a comment?